src/main/scala/io/iohk/ethereum/jsonrpc/EthJsonMethodsImplicits.scala

```diff
@@ -0,0 +1,130 @@
+package io.iohk.ethereum.jsonrpc
+
+import io.iohk.ethereum.jsonrpc.EthService._
+import io.iohk.ethereum.jsonrpc.JsonRpcController.{JsonDecoder, JsonEncoder}
+import io.iohk.ethereum.jsonrpc.JsonRpcErrors.InvalidParams
+import org.json4s.{Extraction, JsonAST}
+import org.json4s.JsonAST.{JArray, JBool, JString, JValue, _}
+import org.json4s.JsonDSL._
+
+object EthJsonMethodsImplicits extends JsonMethodsImplicits {
+
+  implicit val eth_protocolVersion = new JsonDecoder[ProtocolVersionRequest] with JsonEncoder[ProtocolVersionResponse] {
+    def decodeJson(params: Option[JArray]): Either[JsonRpcError, ProtocolVersionRequest] = Right(ProtocolVersionRequest())
+
+    def encodeJson(t: ProtocolVersionResponse): JValue = t.value
+  }
+
+  implicit val eth_blockNumber = new JsonDecoder[BestBlockNumberRequest] with JsonEncoder[BestBlockNumberResponse] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, BestBlockNumberRequest] = Right(BestBlockNumberRequest())
+
+    override def encodeJson(t: BestBlockNumberResponse): JValue = Extraction.decompose(t.bestBlockNumber)
+  }
+
+  implicit val eth_submitHashrate = new JsonDecoder[SubmitHashRateRequest] with JsonEncoder[SubmitHashRateResponse] {
+    override def decodeJson(params: Option[JsonAST.JArray]): Either[JsonRpcError, SubmitHashRateRequest] = params match {
+      case Some(JArray((hashRate: JString) :: (id: JString) :: Nil)) =>
+        tryExtractQuantity(hashRate)
+          .flatMap(h => tryExtractUnformattedData(id).map(i => (h, i)))
+          .map { case (h, i) => SubmitHashRateRequest(h, i) }
+      case _ =>
+        Left(InvalidParams)
+    }
+
+    override def encodeJson(t: SubmitHashRateResponse): JValue = JBool(t.success)
+  }
+
+  implicit val eth_getWork = new JsonDecoder[GetWorkRequest] with JsonEncoder[GetWorkResponse] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, GetWorkRequest] = params match {
+      case None => Right(GetWorkRequest())
+      case Some(_) => Left(InvalidParams)
```

Comment:
Geth allows any number of arguments for parameterless methods - it simply ignores the `params` field. I believe we should at least allow `params` to be empty (but ignoring is fine too).